ocaml/typing/subst.ml

```diff
@@ -61,15 +65,43 @@ let add_modtype_path p ty s =
   { s with modtypes = Path.Map.add p ty s.modtypes; last_compose = None }
 let add_modtype id ty s = add_modtype_path (Pident id) ty s
 
-let for_saving s = { s with for_saving = true; last_compose = None }
+let for_saving s =
+  (* CR layouts: it would be better to put all this stuff outside this
+     function, but it's in here because we really want to tailor the reason
+     to describe the module a symbol is imported from. But RAE's initial
+     attempt to do this based on filename caused spurious "inconsistent
+     assumption" errors that couldn't immediately be solved. Revisit
+     with a better approach. *)
```

Comment:
I would love to say that `t has layout value, because it was imported with that layout from foo.ml`. It's true that the information might not be necessary, but `t has layout value, because it was imported with that layout` seems not as great. In any case, I suspect we'll have to revisit all of this later.